src/model/implementation/field.ts

```diff
@@ -1687,6 +1687,20 @@ export class Field implements ModelComponent {
                 ),
             );
         }
+        if (
+            this.isIncludedInSearch &&
+            this.type.isObjectType &&
+            !this.type.fields.some(
+                (value) => value.isIncludedInSearch,
+            )
+        ) {
```

Comment:
how does this work for `isFulltextIncludedInSearch`? You probably should annotate the objects with `@flexSearch(includeInSearch: true)` regardless of fulltext/not fulltext, right? But then the `.some` should probably also allow `isFulltextIncludedInSearch`.